### ISSUE TITLE
Directory Structure

### ISSUE THREAD

**Issue by ThunderSon** [OWASP/www-project-top-ten#5]
**What is wrong?**
`www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A1-Injection` is too repetitive.

**What is a solution?**
Update the structure to something like the following:
`www-project-top-ten/2017/A1-Injection`

**How could it be done?**
Update the file names to exactly what they are, `AX-<what it is>`
Update the folder name to follow the above mentioned nomenclature.

**Comment by kingthorin** [comment]
Unfortunately this likely means a redirect needs to be implemented for other things already linking with the giant links :disappointed:

**Comment by sslHello** [comment]
@kingthorin Thank you for your suggestion.
As the migtaion team has not asked the project team while migrating, we found this result as you did, too.
Do you know any simple way to generate redirects in Github Pages using Jekyll?
If there is a technical simple way we are happy to discuss this. For now we won't like to prodice broken links....
Cheers Torsten

**Comment by kingthorin** [comment]
I don’t know if a good way aside from putting placeholder pages with meta redirects.

**Comment by kingthorin** [comment]
I’ll look into this a bit more and let you know if I come up with something.

**Comment by kingthorin** [comment]
Here's a pretty simple solution:

```diff
diff --git a/2017/A1-Injection.md b/2017/A1-Injection.md
index 104d77a..113eee2 100644
--- a/2017/A1-Injection.md
+++ b/2017/A1-Injection.md
@@ -11,6 +11,7 @@ exploitability: 3
 prevalence: 2
 detectability: 3
 technical: 3
+redirect_from: OWASP_Top_Ten_2017/Top_10-2017_A1-Injection.html
 
 ---
 
diff --git a/_config.yml b/_config.yml
index 5dd48ce..401e213 100644
--- a/_config.yml
+++ b/_config.yml
@@ -1,3 +1,4 @@
 remote_theme: "owasp/www--site-theme"
 plugins:
- - jekyll-include-cache-0.2.0
\ No newline at end of file
+ - jekyll-include-cache-0.2.0
+ - jekyll-redirect-from
```

`redirect_from` will need to be established on all the moved pages front mater but it isn't that big a deal (probably even automateable).

Refs:
- https://help.github.com/en/enterprise/2.13/user/articles/redirects-on-github-pages
- https://github.com/jekyll/jekyll-redirect-from

Note:
1) If necessary multiple redirect_from values can be specified as:
```yml
redirect_from:
  -  /foo/a1
  - OWASP_Top_Ten_2017/Top_10-2017_A1-Injection.html
```
2) I have tested locally based on @ThunderSon's PR branch: https://github.com/OWASP/www-project-top-ten/pull/6 (Using these instructions but with the appropriate repo/branch checked out: https://github.com/OWASP/www-project-web-security-testing-guide/#getting-started)

Basically at runtime it builds/renders the structure of all the redirect_from files/directories and using link rel canonical along with meta refresh. From the little reading I've done it is search engine friendly and they'll pickup the updates after a few days (or so I read).

**Comment by ThunderSon** [comment]
That looks fantastic! I'll do some testing tomorrow in my free time.

**Comment by sslHello** [comment]
Hi , thank you for your research and your tests.
I've checked that the plugin is whitelisted, too: https://pages.github.com/versions/ :-)
We are going to have our next project meeting next Monday, evening.
I am happy to discuss and test it.
For the names of the pages it might be that we'd support to keep the year or more as we have intentionally added the year to A1:2017 to A10:2017 in the Pdf. This has been to prevent people from referencing/using A5 and nobody knows from which version ....
The webpages sould link now relatevely to each oher, so people could also copy the files locally.
Thanks a lot for now.
Cheers Torsten

**Comment by kingthorin** [comment]
The file names shouldn't need the year reference as the live in the year folder? Or were you more concerned about the titles?

**Comment by ThunderSon** [comment]
We followed an API naming convention, or a general good naming convention if you want.
OWASP -> Top Ten -> 2017 -> A1-A10; that should be enough, don't you think? The year is needed to access the controls being referenced.

**Comment by sslHello** [comment]
Hi again,
we have discussted your suggestion in our latest two meetings:
We are happy to get shorter file names and to get rid of the doubled encoding for "Logging&Monitoring". As we see it manages to keep the actual links working while moving to new names, great  :-).
As I took into account last time, we have discussed also your view about
how to migrate A1 to A10, e.g. www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A1-Injection. 
Finally we have decided that we prefer to keep the year for A1_2017 to A10_2017 in the fime names even if it is already in the folder name, e.g. www-project-top-ten/2017/A1_2017-Injection. This to follow the additional year in the haedlines of the pptx/pdf version. We hope that people reference the individual Top 10 risk includung the year, e.g. A1_2017.
I am happy to test it in a fork.
Thank you very much for your suggestion.

**Comment by kingthorin** [comment]
Well even going from `www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A1-Injection.` to `www-project-top-ten/2017/2017_A1-Injection.` would be an improvement.

**Comment by ThunderSon** [comment]
I'll tackle this the soonest. Thanks for discussing this.

**Comment by sslHello** [comment]
Many thanks to @ThunderSon and @kingthorin;
Commit 37014c4 based on this pull request #6 and should solve this issue
.... sorry for the delay... 
Cheers Torsten
